codebase/codebase.go

```diff
@@ -0,0 +1,68 @@
+package codebase
+
+import "errors"
+
+// Codebase defines all parameters those are useful to associate Che Editor's window to a WI
+type Codebase struct {
+	Repository string `json:"repository"`
+	Branch     string `json:"branch"`
+	FileName   string `json:"filename"`
+	LineNumber int    `json:"linenumber"`
+}
```

Comment:
Update : Talked to Ilya and got [this swagger API link](http://che-starter.prod-preview.openshift.io/swagger-ui.html#!/workspace-controller/createUsingPOST) 
@aslakknutsen Can we finalize attributes needed for this API ?
I could see `config`, `login` like important attributes. Will those be handled by `codebase API` ?